### ISSUE TITLE
csc is unable to locate the extension assembly in dependency graph with v3.5.2

### ISSUE THREAD

**Issue by am11** [Azure/azure-sdk-for-media-services-extensions#31]
Minimum steps to repro:

In Windows cmd with dotnet-cli in PATH:

```cmd
mkdir \temp\test
cd \temp\test
dotnet new
```

Edit `project.json` and replace the contents with:

```json
{
  "frameworks": { "net451": {} },
  "dependencies": { "windowsazure.mediaservices.extensions": "3.5.2" }
}
```

Edit `Program.cs` and replace the contents with:

```cs
using Microsoft.WindowsAzure.MediaServices.Client; 

public class Program {
  public static void Main() {
    IAsset asset = null;
    asset.GetHlsUri();
  }
}
```

In cmd, running `dotnet restore && dotnet build` yields:

```cmd
C:\Program Files\dotnet\dotnet.exe compile-csc @C:\temp\test\obj\Debug\net451\dotnet-compile.rsp returned Exit Code 1
C:\temp\test\Program.cs(6,11): error CS1061: 'IAsset' does not contain a definition for 'GetHlsUri' and no extension method 'GetHlsUri' accepting a first argument of type 'IAsset' could be found (are you missing a using directive or an assembly reference?)

Compilation failed.
    0 Warning(s)
    1 Error(s)
```
(i.e. `IAsset` is recognized but not the extension method `GetHslUri()`)

Replace `"3.5.2"` with `"3.3.0"` in `project.json` and run `dotnet restore && dotnet build` to get:

```cmd
Compilation succeeded.
    0 Warning(s)
    0 Error(s)
```

**Comment by johndeu** [comment]
Roll back to version v3.3.0. There is a broken build right now in 3.5.2. 
We are working on getting the build fixed.

**Comment by mconverti** [comment]
@am11 please try again with the latest version of the windowsazure.mediaservices.extensions nuget package (v3.7.0.1). The issue should be fixed now.

**Comment by am11** [comment]
I can confirm that the issue is fixed with v3.7.0.1. Thanks @mconverti! 👍 
Would be nice to mark the broken version 3.5.2 deprecated, so folks relying on that get the warning.
